MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -210,12 +212,16 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
     }
     
     func updateArrowAnnotations(_ routeProgress: RouteProgress) {
-        guard routeProgress.currentLegProgress.upComingStep != nil else {
+        guard let step = routeProgress.currentLegProgress.upComingStep else {
             return
         }
         
-        mapView.removeArrow()
-        mapView.addArrow(routeProgress)
+        if step != arrowCurrentStep {
+            mapView.removeArrow()
+            mapView.addArrow(routeProgress)
```

Comment:
Instead of tearing down all the style layers and sources then rebuilding them from scratch, we should simply update the source’s `shape` property to reflect the current step. The check above becomes unnecessary, because mbgl will automatically redraw the style layers without any flashing.